### DISCUSSION TITLE
How to run telegram-chat-forward on Mac ?

### DISCUSSION THREAD

**Discussion by alanliew88** [aahnik/tgcf#25] [Q&A] [answered]
1. I use the python anywhere and found out that they dont support some of the telethon fucntion.
2.  Use mac to run the code, however, mac does not support .env file. How to run it ? any idea ?

**Comment by aahnik** [comment, marked as answer]
@alanliew88 , yes, pythonanywhere free version does not support mtproto connections.


- Mac supports `.env` files. You need to use the terminal.

- Open your terminal and create a file named `.env` and edit it.

- You can use the text editor named `micro` . (you can use any text editor)

     - Install `micro` using homebrew (`brew install micro`)
     - Now move into the folder of `telegram-chat-forward`
     - Run `micro .env`
     - Start writing
     - After you are done, save and exit. (Ctrl/Cmd + S and Ctrl/Cmd + Q)

**Reply by aahnik** [reply]
@alanliew88 please join the telegram channel to get updates. 

Please join the **telegram channel** [tg_cf](https://telegram.me/tg_cf). Here I will post updates about telegram-chat-forward , without notifying you.

Only breaking changes will be notified.

Please do not mute the channel. I will post a message only when there is a change in code.

**Comment by alanliew88** [comment]
<img width="949" alt="Screenshot 2021-03-01 at 12 41 47 AM" src="https://user-images.githubusercontent.com/79797236/109426180-32d40100-7a27-11eb-8769-2ed6603f8613.png">
ValueError: Could not find the input entity for <telethon.tl.types.PeerUser object at 0x102c8e4f0>. Please read https://docs.telethon.dev/en/latest/concepts/entities.html to find out more details.



May I know how to solve this ? I have been logged in and paste the code i received in telegram

**Reply by aahnik** [reply]
moved to https://github.com/aahnik/telegram-chat-forward/discussions/26
